rasa/graph_components/validators/config_validator.py

```diff
@@ -0,0 +1,514 @@
+    Returns:
+        text containing all type names
+    """
+    return ", ".join([type.__name__ for type in types])
+
+
+class DefaultV1ConfigValidator(GraphComponent):
+    """Validates the current graph schema against the training data and domain.
+
+    This validator is just a replacement for the configuration checks done in Rasa 2.x
+    and should only be used for "default v1" configurations.
+    E.g. it just checks whether a tokenizer exists somewhere in the given graph schema,
+    but it does not check whether the tokenizer appears on the right paths.
+    """
+
+    @classmethod
+    def create(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> DefaultV1ConfigValidator:
+        """Creates a new `ConfigValidator` (see parent class for full docstring)."""
+        return cls(execution_context.graph_schema)
+
+    def __init__(self, graph_schema: GraphSchema) -> None:
+        """Instantiates a new `ConfigValidator`.
+
+        Args:
+           graph_schema: a graph schema
+        """
+        self._graph_schema = graph_schema
+        self._component_types = set(node.uses for node in graph_schema.nodes.values())
+        self._policy_schema_nodes: List[SchemaNode] = [
+            node
+            for node in self._graph_schema.nodes.values()
+            if issubclass(node.uses, PolicyGraphComponent)
+        ]
+
+    def validate(self, importer: TrainingDataImporter) -> TrainingDataImporter:
+        """Validates the current graph schema against the training data and domain.
+
+        Args:
+            importer: the training data importer which can also load the domain
+        """
+        nlu_data = importer.get_nlu_data()
+        self._validate_nlu(nlu_data)
+
+        story_graph = importer.get_stories()
+        domain = importer.get_domain()
+        self._validate_core(story_graph, domain)
+        return importer
+
+    def _validate_nlu(self, training_data: TrainingData) -> None:
+        """Validates whether the configuration matches the training data.
+
+        Args:
+           training_data: training_data
+        """
+        self._raise_if_more_than_one_tokenizer()
+        self._raise_if_featurizers_are_not_compatible()
+        self._warn_of_competing_extractors()
+        self._warn_of_competition_with_regex_extractor(training_data=training_data)
+        self._warn_if_some_training_data_is_unused(training_data=training_data)
+
+    def _warn_if_some_training_data_is_unused(
+        self, training_data: TrainingData
+    ) -> None:
+        """Validates that all training data will be consumed by some component.
+
+        For example, if you specify response examples in your training data, but there
+        is no `ResponseSelector` component in your configuration, then this method
+        issues a warning.
+
+        Args:
+            training_data: training data
```

Comment:
There is not much to say about `training_data` but just repeating the param name is not ideal in my opinion. How about something like this?
```suggestion
            training_data: The training data for the NLU components.
```